src/nni_manager/training_service/pai/paiTrainingService.ts

```diff
@@ -324,18 +324,19 @@ class PAITrainingService implements TrainingService {
                 "Authorization": 'Bearer ' + this.paiToken
             }
         };
+
         request(stopJobRequest, (error: Error, response: request.Response, body: any) => {
             if (error || response.statusCode >= 400) {
                 this.log.error(`PAI Training service: stop trial ${trialJobId} to PAI Cluster failed!`);
                 deferred.reject(error ? error.message : 'Stop trial failed, http code: ' + response.statusCode);                
             } else {
-                if (isEarlyStopped) {
-                    trialJobDetail.status = 'EARLY_STOPPED';
-                }
                 deferred.resolve();
             }
         });
 
+        // Set trialjobDetail's early stopped field, to mark the job's cancellation source
+        trialJobDetail.isEarlyStopped = isEarlyStopped;
```

Comment:
I think this line has to be put at the beginning of this function.

Reply:
Not very sure why we should put this line to the beginning. 
The beginning of this function has many condition checking. If we put this line to the beginning of this function but condition checking failed, the trial job status may be wrong.

Reply:
agree, then we should put this line before line 328 `request(...`

Reply:
Yes Sure.